samples/nrf_desktop/src/hw_interface/buttons.c

```diff
@@ -62,15 +84,19 @@ static int set_cols(u32_t mask)
 
 static int get_rows(u32_t *mask)
 {
-	for (size_t i = 0; i < ARRAY_SIZE(col_pin); i++) {
+	for (size_t i = 0; i < ARRAY_SIZE(col); i++) {
 		u32_t val;
 
-		if (gpio_pin_read(gpio_dev, row_pin[i], &val)) {
+		if (gpio_pin_read(gpio_devs[row[i].port], row[i].pin, &val)) {
 			LOG_ERR("Cannot get pin");
 			return -EFAULT;
 		}
 
-		(*mask) |= (val << i);
+		if (IS_MATRIX) {
+			(*mask) |= ((val) << i);
```

Comment:
I am not entirely sure why this depends on being a matrix... maybe we should instead have a button pin polarity defined (I guess it could be one config for all pins).